apiserver/params/params.go

```diff
@@ -768,37 +768,97 @@ type RebootActionResult struct {
 
 const (
-	Alive = Life(multiwatcher.Alive)
-	Dying = Life(multiwatcher.Dying)
-	Dead  = Life(multiwatcher.Dead)
+	Alive Life = "alive"
+	Dying Life = "dying"
+	Dead  Life = "dead"
 )
 
 // Status represents the status of an entity.
 // It could be a unit, machine or its agent.
 type Status multiwatcher.Status
 
+// TranslateLegacyStatus returns the status value clients expect to see for Juju 1.x.
+func TranslateLegacyStatus(in Status) Status {
+	switch in {
+	case StatusFailed:
+		return StatusDown
+	case StatusActive:
+		return StatusStarted
+	default:
+		return in
+	}
+}
+
 const (
-	// The entity is not yet participating in the environment.
-	StatusPending = Status(multiwatcher.StatusPending)
+	// Status values common to machine and unit agents.
 
-	// The unit has performed initial setup and is adapting itself to
-	// the environment. Not applicable to machines.
-	StatusInstalled = Status(multiwatcher.StatusInstalled)
+	// The entity requires human intervention in order to operate
+	// correctly.
+	StatusError Status = "error"
 
 	// The entity is actively participating in the environment.
-	StatusStarted = Status(multiwatcher.StatusStarted)
+	// For unit agents, this is a state we preserve for backwards
+	// compatibility with scripts during the life of Juju 1.x.
+	// In Juju 2.x, the agent-state will remain “active” and scripts
+	// will watch the unit-state instead for signals of service readiness.
+	StatusStarted Status = "started"
+)
 
-	// The entity's agent will perform no further action, other than
-	// to set the unit to Dead at a suitable moment.
-	StatusStopped = Status(multiwatcher.StatusStopped)
+const (
+	// Status values specific to machine agents.
 
-	// The entity requires human intervention in order to operate
-	// correctly.
-	StatusError = Status(multiwatcher.StatusError)
+	// The machine is not yet participating in the environment.
+	StatusPending Status = "pending"
 
-	// The entity ought to be signalling activity, but it cannot be
+	// The machine's agent will perform no further action, other than
+	// to set the unit to Dead at a suitable moment.
+	StatusStopped Status = "stopped"
+
+	// The machine ought to be signalling activity, but it cannot be
 	// detected.
-	StatusDown = Status(multiwatcher.StatusDown)
+	StatusDown Status = "down"
+)
+
+const (
+	// Status values specific to unit agents.
+
+	// The machine on which a unit is to be hosted is still being
+	// spun up in the cloud.
+	StatusAllocating Status = "allocating"
+
+	// The unit agent is downloading the charm and running the install hook.
+	StatusInstalling Status = "installing"
+
+	// The agent is actively participating in the environment.
+	StatusActive Status = "active"
+
+	// The unit is being destroyed; the agent will soon mark the unit as “dead”.
+	// In Juju 2.x this will describe the state of the agent rather than a unit.
+	StatusStopping Status = "stopping"
+
+	// The unit agent has failed in some way,eg the agent ought to be signalling
```

Comment:
Perhaps also mention that this status could be because the unit agent has signalled a fatal error? (as per the spec)